provider/ec2/environ.go

```diff
@@ -1379,17 +1378,18 @@ func mapNetworkInterface(iface amzec2.NetworkInterface, subnet amzec2.Subnet) ne
 			iface.PrivateIPAddress,
 			network.WithScope(network.ScopeCloudLocal),
 			network.WithCIDR(subnet.CIDRBlock),
+			network.WithConfigType(network.ConfigDHCP),
 		)},
 		Origin: network.OriginProvider,
 	}
 
 	for _, privAddr := range iface.PrivateIPs {
 		if privAddr.Association.PublicIP != "" {
-			ni.ShadowAddresses = append(
-				ni.ShadowAddresses,
-				network.NewProviderAddress(
-					privAddr.Association.PublicIP, network.WithScope(network.ScopePublic)),
-			)
+			ni.ShadowAddresses = append(ni.ShadowAddresses, network.NewProviderAddress(
+				privAddr.Association.PublicIP,
+				network.WithScope(network.ScopePublic),
+				network.WithConfigType(network.ConfigDHCP),
```

Comment:
Probably doesn't change anything but why do we assume that ec2 assigns VIPs via DHCP?

Reply:
I don't know, and it's probably incorrect anyway. If it was correct, your VM could change public IP during downtime, which seems like undesirable behaviour.

Reply:
As soon as I can manage, I will change the instance-poller API facade to process these addresses. I'll look at this properly at that time.

Reply:
That seems to be the case on ec2 at least. See https://github.com/juju/juju/pull/10902 and associated bug.

Reply:
Yes, now that I think on it, they would rather charge for assignable floating IPs than give you the "feature" for free.